kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaChanges.java

```diff
@@ -35,61 +36,75 @@
  * <p>ToDo: Possibly track moves/renames independently, enable capturing re-ordered columns in top
  * level schema
  */
-class SchemaChanges {
-  private List<StructField> addedFields;
-  private List<StructField> removedFields;
-  private List<Tuple2<StructField, StructField>> updatedFields;
+class SchemaChanges<T extends Supplier<StructField>> {
+  private final List<T> addedFields;
+  private final List<T> removedFields;
+  private final List<T> renamedFields;
```

Comment:
I separated out fields which *only* went through renames into renamed fields as opposed to including those in updatedFields. The reason I did it was because the validation for if a map key was modified needs to allow for renames as well. To distinguish if two fields are only different by name I added DataType#equalsIgnoringNames. See those changes below.